### PR TITLE
Create and mount service CA via ConfigMap

### LINKED ISSUE #1123: Create and mount CA bundle via ConfigMap
Starting from OpenShift 4.6, the CA bundle will only be available via ConfigMap, as described here: 
https://docs.openshift.com/container-platform/4.4/authentication/certificates/service-serving-certificate.html#add-service-certificate-configmap_service-serving-certificate

We currently use this when setting the flag `reporter.grpc.tls.ca`: all-in-one deployment, daemonset agent and sidecar agent.

### PR DESCRIPTION
**Pull request by jpkrohling**
Fixes #1123

### CODE REVIEW

File: pkg/config/ca/ca.go [hunk 1]
```diff
@@ -55,6 +64,47 @@ func GetTrustedCABundle(jaeger *v1.Jaeger) *corev1.ConfigMap {
 	}
 }
 
+// GetServiceCABundle returns a trusted CA bundle configmap if platform is OpenShift
```
Comment (objectiser) [inline pkg/config/ca/ca.go:67]: nit: "a service CA bundle..."

File: pkg/config/ca/ca.go [hunk 2]
```diff
@@ -55,6 +64,47 @@ func GetTrustedCABundle(jaeger *v1.Jaeger) *corev1.ConfigMap {
...
+func GetServiceCABundle(jaeger *v1.Jaeger) *corev1.ConfigMap {
+	// Only configure the trusted CA if running in OpenShift
```
Comment (objectiser) [inline pkg/config/ca/ca.go:69]: same trusted->service?

File: pkg/config/ca/ca.go (hunk 1 above)
Comment (jpkrohling) [reply pkg/config/ca/ca.go]: Fixed.

File: pkg/config/ca/ca.go (hunk 2 above)
Comment (jpkrohling) [reply pkg/config/ca/ca.go]: Fixed!

**Pull request merged**